### ISSUE TITLE
The supplement about the answer of Exercise 6 in Lab1.

### ISSUE THREAD

**Issue by loggerhead** [petroav/6.828#1]
I can't find answers of the following questions, so I give the answer of mine. Hope we could discuss it's right or wrong 😄 

> Why are they different?

At the point the BIOS enters the boot loader, I am agree with you, but I have found that the content don't changed but always `0x00000000`; At the point the boot loader enters the kernel, because `readseg` has loaded the segment started at `0x00100000`, so the 8 words of memory at 0x00100000 should be the first 8 words of this segment.

> What is there at the second breakpoint?

The 8 words keeps unchanged after the loop in `bootmain` until entering the kernel.

```
0x1badb002  0x00000000  0xe4524ffe  0x7205c766
0x34000004  0x0000b812  0x220f0011  0xc0200fd8
```

**Comment by petroav** [comment]
@loggerhead apologies for the late reply!

Yes I agree with everything you say and my answer could definitely use some clarification.

I didn't actually run the debugger for this exercise and tried to reason about the answer. I now ran GDB and indeed before loading the kernel the 8 words starting at `0x00100000` are always set to `0x00000000` and don't change between machine restarts. In my answer I wanted to convey that you have no guarantee about the contents of that memory location. Memory content at startup is determined by the hardware. So I think it just happens that QEMU works in such a way that memory in that location is set to `0x00000000` on restart. That might differ to how a physical machine would behave.

As for the second breakpoint you also have the right memory content and in fact if you look closely the memory content matches the instruction binary you paste above (albeit the two have different endianness).

I'm glad you found this repo useful and let me know if you want to discuss anything else from it!

**Comment by loggerhead** [comment]
> Memory content at startup is determined by the hardware. So I think it just happens that QEMU works in such a way that memory in that location is set to 0x00000000 on restart. That might differ to how a physical machine would behave.

I don't think this deeply, but you remind me. In my opinion, QEMU will more likely to simulate exactly what the physical machine does, so the zeros in memory is not a special coincidence but intented. And I found [a question in quora](https://www.quora.com/When-RAM-is-powered-up-does-it-contain-garbage-values-or-zeroes) gives some ideas:

1. From the answer of Dan Zhang, we can infer that if no instruction involved, then the value of memory after power-up should be all zeros or all ones.
2. From the answer of Ron Garrett, we know that the value of memory probably changed be BIOS test  routine or something related to.

All in all, I think the initialzation of memory are certain values.

> As for the second breakpoint you also have the right memory content and in fact if you look closely the memory content matches the instruction binary you paste above (albeit the two have different endianness).

Yes, I know. I just lazy to type it down :P 

> I'm glad you found this repo useful and let me know if you want to discuss anything else from it!

Yes, your repo helped me a lot, thanks!
